oauthproxy.go

```diff
@@ -277,6 +282,53 @@ func buildSignInMessage(opts *options.Options) string {
 	return msg
 }
 
+// buildRoutesAllowlist builds an []allowedRoute  list from either the legacy
+// SkipAuthRegex option (paths only support) or newer SkipAuthRoutes option
+// (method=path support)
+func buildRoutesAllowlist(opts *options.Options) ([]*allowedRoute, error) {
+	routes := make([]*allowedRoute, 0, len(opts.SkipAuthRegex)+len(opts.SkipAuthRoutes))
+
+	for _, path := range opts.SkipAuthRegex {
+		compiledRegex, err := regexp.Compile(path)
+		if err != nil {
+			return nil, err
+		}
+		logger.Printf("Skipping auth - Method: ALL | Path: %s", path)
+		routes = append(routes, &allowedRoute{
+			method:    "",
+			pathRegex: compiledRegex,
+		})
+	}
+
+	for _, methodPath := range opts.SkipAuthRoutes {
+		var (
+			method string
+			path   string
+		)
+
+		parts := strings.Split(methodPath, "=")
```

Comment:
There's 1 case where this logic won't be fully backwards compatible with migrating former `--skip-auth-regex` entries to this without a method:

Regexes with a `=` in it (e.g querystring) so `/my/path\?q=data` will view `/my/path\?q` as the HTTP method to allow and `data` as the path regex.

The hack for this scenario is `=/my/path\?q=data`.  Just isn't graceful. Forcing a regex wrapper (E.g. `/regex/` or `#regex#` etc) would work? But would make the migration harder for 99% of use cases.

I lean toward leave the corner case not fully compatible, understand its hacky because we are using single configs, and know this will be better with structured configs.

Reply:
Maybe worth noting this in the changelog?